Plugins/GitUIPluginInterfaces/BuildServerIntegration/BuildInfo.cs

```diff
@@ -16,10 +16,26 @@ public enum BuildStatus
 
         public string Id { get; set; }
         public DateTime StartDate { get; set; }
+        public long? Duration { get; set; }
         public BuildStatus Status { get; set; }
         public string Description { get; set; }
         public string[] CommitHashList { get; set; }
         public string Url { get; set; }
         public bool ShowInBuildReportTab { get; set; } = true;
+
+        public string DurationText()
+        {
+            if (Duration.HasValue)
+            {
+                return " (" + (TimeSpan.FromMilliseconds(Duration.Value)).ToString(@"mm\:ss") + ")";
+            }
+
+            return string.Empty;
+        }
+
+        public void AddDurationToDescription()
```

Comment:
Same as above.
I would remove it completely, as it is only used in `JenkinsAdapter` and modify the adapter to set the `Description` the way it requires.